packages/client/tests/functional/issues/15644-middleware-arg-types/tests.ts

```diff
@@ -0,0 +1,183 @@
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+testMatrix.setupTestSuite(
+  (suiteConfig, suiteMeta) => {
+    test('middleware with count', async () => {
+      prisma.$use((params, next) => {
```

Comment:
```suggestion
    test('middleware with count', async () => {
      expect.assertions(1)
      prisma.$use((params, next) => {
```

Reply:
Let's use `expect.assertions` to ensure that the assertion happens in the callback.